cs3api4lab/api/cs3_share_api.py

```diff
@@ -37,12 +39,15 @@ class Cs3ShareApi:
 
     def __init__(self, log):
         self.log = log
-        config = Cs3ConfigManager.get_config()
-        channel = ChannelConnector.get_channel()
-        self.config = config
-        self.auth = Authenticator(config, channel)
+        self.config = Cs3ConfigManager().get_config()
+        self.auth = Authenticator(config=self.config, log=self.log).instance
+
         self.file_api = Cs3FileApi(log)
-        self.cs3_api = grpc_gateway.GatewayAPIStub(channel)
+
+        channel = ChannelConnector().get_channel()
```

Comment:
Why don't we already return the intercepted channel in `ChannelConnector().get_channel()`?

Reply:
Because, channels it is a generic type using for authentication, .etc, intercept channel is specific type using in Riva data operation (stat, read a file, write a file, .etc).

Reply:
Ok, understood

Reply:
Ok, makes sense.